### ISSUE TITLE
Useful overview disappears, history takes ages and/or Craft hangs  / freezes when history or review is requested...

### ISSUE THREAD

**Issue by MattAppleton** [dispositiontools/craft-alt-text-generator#9]
When you first fire up Alt Text Generator you get a super-useful bit of feedback saying how many images haven't got alt text, how many cannot be processed (for example they might be too big) etc etc.

However, once the plugin starts to process things this overview disappears!

Accessing the history is VERY slow, and seems to freeze Craft altogether (this is on a site that had maybe 1k images processed).

Clicking on 'view items to review' gives me a major hang and a 'service unavailable' message!

<img width="531" alt="Mica_Screenshot 2024-10-08 at 19 37 45" src="https://github.com/user-attachments/assets/d6fdbaae-13d4-4159-86d7-7d3c1772b8d6">

perhaps this is to do with lots of thumbnails overwhelming the system, but something is not built right here!

Perhaps one idea might be to give text-only feedback on larger bodies of data, or simply top-level numbers at first.

**Comment by MattAppleton** [comment]
I'm running Craft 4.12.5 with this install...

**Comment by dispositiontools** [comment]
@MattAppleton I'm really sorry about this issue - I'm going to take a look at this and do some testing - I def don't want it to cause a hang! and i'm sorry for the delay in replying - I've been away for a combined family wedding and holiday

**Comment by MattAppleton** [comment]
@dispositiontools  — thanks for the update...

**Comment by dispositiontools** [comment]
@MattAppleton Are you able to share with me your server specs for this?

**Comment by MattAppleton** [comment]
Sure... 

Our Craft installs all run on similar setup, though soem are now Craft 5. This particular setup is


PHP version	8.3.0
OS version	Linux 6.1.10-x86_64-linode159
Database driver & version	MySQL 8.0.39
Image driver & version	GD 8.3.0
Craft edition & version	Craft Pro 4.12.5
Yii version	2.0.51
Twig version	v3.14.0
Guzzle version	7.9.2

Running on Linode with Serverpilot app layer.

Some key PHP info below. Is there any other information you might need?

max_execution_time	30
max_file_uploads	20
max_input_nesting_level	64
max_input_time	-1
max_input_vars	5000
max_multipart_body_parts	-1
memory_limit	256M
open_basedir	no value
output_buffering	0
output_encoding	no value
output_handler	no value
post_max_size	1024M
precision	14
realpath_cache_size	4096K
realpath_cache_ttl	120
register_argc_argv	On
report_memleaks	On
report_zend_debug	Off

**Comment by dispositiontools** [comment]
Hi @MattAppleton 

this has been really helpful, and I've diagnose the issue and the next release will have some options to help with this. When we do an asset query with no limits and the process what needs to happen it actually puts quite a lot in to memory. thanks for bringing this to my attention

**Comment by MattAppleton** [comment]
thanks — so for now theres no change, is that right?

**Comment by dispositiontools** [comment]
Yes, not changes for those views just yet - but very soon.

**Comment by darylknight** [comment]
Any news on this? History view just took about 30 seconds to load for me

**Comment by dispositiontools** [comment]
I've pushed updates v4.2.1 and v5.2.1 with improvements to the pages. I've added pagination and update the view. 
More work to come, but this should be some initial help.
